src/client/Microsoft.Identity.Client/MsalManagedIdentityException.cs

```diff
@@ -48,5 +83,26 @@ public class MsalManagedIdentityException : MsalServiceException
         /// Specifies the managed identity source from which the exception initiates.
         /// </summary>
         public ManagedIdentitySource ManagedIdentitySource { get; }
+
+        /// <summary>
+        /// Retry error codes specific to managed identity
+        /// </summary>
+        protected override void UpdateIsRetryable()
+        {
+            switch (StatusCode)
+            {
+                case 404: //Not Found
```

Comment:
404 should not be retriable?

Reply:
I think it is. Some scenario where IMDS is getting updated.

Reply:
Yes, 404 is specifically retried for managed identity scenarios in azure identity repo. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/ManagedIdentitySource.cs#L153